internal/api/core/core.go

```diff
@@ -47,7 +48,7 @@ func New(conf config.Config, banner string) (*app.Runner, dependency.Persistence
 	}
 	persesAPI := NewPersesAPI(serviceManager, persistenceManager, conf)
 	persesFrontend := ui.NewPersesFrontend()
-	runner := app.NewRunner().WithDefaultHTTPServer("perses").SetBanner(banner)
+	runner := app.NewRunner().WithDefaultHTTPServerAndPrometheusRegisterer(metricNamespace, registry, registry).SetBanner(banner)
```

Comment:
Now I'm realising that I might have been wrong in my perses/common review this is actually a registry and not registerer... Btw why are you giving it twice there ?

Reply:
that's because I had to modify (again) the `perses/common` lib to add another parameter called `Gatherer`.
Here the prometheus registry is implementing the `Register` and the `Gatherer` interface. And it seems if you provide a different object of each interface, the metrics doesn't appear in `/metrics`.

It's really weird we need to do that. But meh

Reply:
well that's what I have observed, but if there is two different interfaces, that probably means you can provide two different implementations. I just didn't get then how to use them.
So for the moment I prefer to let `perses/common` the possibility to expose both interface.